src/extensions/magic_mount/chunk.rs

```diff
@@ -0,0 +1,116 @@
+use async_recursion::async_recursion;
+
+const CHUNK_SIZE: usize = 1024 * 1024; // use 1MB chunks
+
+type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+impl crate::application::CoLink {
+    #[async_recursion]
+    pub(crate) async fn _create_entry_chunk(
+        &self,
+        key_name: &str,
+        payload: &[u8],
+    ) -> Result<String, Error> {
+        let metadata_key = format!("{}:chunk_metadata", key_name);
+
+        // lock the metadata entry to prevent simultaneous writes
+        self.create_entry(
+            &metadata_key.clone(),
+            &("creation-in-progress-locked".to_string().into_bytes()),
+        )
+        .await?;
+
+        // create the chunks and store them
+        let mut offset = 0;
+        let mut chunk_id = 0;
+        let mut chunk_paths = Vec::new();
+        while offset < payload.len() {
+            let chunk_size = if offset + CHUNK_SIZE > payload.len() {
+                payload.len() - offset
+            } else {
+                CHUNK_SIZE
+            };
+            let response = self
+                .create_entry(
+                    &format!("{}:{}", key_name, chunk_id),
+                    &payload[offset..offset + chunk_size],
+                )
+                .await?;
+            chunk_paths.push(response);
```

Comment:
Is there a way to reduce the length of the element in chunk_paths since they share the same prefix? It might be a limit for creating Gigabyte-level chunked entries.

Reply:
we can use timestamp onlyl